op-node/rollup/driver/state.go

```diff
@@ -361,6 +371,9 @@ func (s *state) eventLoop() {
 				s.l2Finalized = finalized
 				s.l2SafeHead = safe
 				s.l2Head = unsafe
+				s.metrics.SetHead("l2_finalized", finalized.Number)
+				s.metrics.SetHead("l2_safe", safe.Number)
+				s.metrics.SetHead("l2_unsafe", unsafe.Number)
```

Comment:
Can you move this up, next to the log statement that only fires on changes? This code path is hit a lot repeatedly with the same values, and I'm worried the metrics collection will slow things down.